### ISSUE TITLE
On first start Playlist is not snapped

### ISSUE THREAD

**Issue by Ryrynz** [Garteal/MPDN_Extensions#2]
I think most people would prefer the playlist snapped to the right of the player so do this by default. I'd also recommend remembering the playlist size and position by default also.

**Comment by Garteal** [comment]
If I interpret you correctly... thus have the playlist stick to the right-side of the player by default (and not the screen), it will be an issue for people with smaller resolutions.

**Comment by Ryrynz** [comment]
Do a check to determine if there's enough space to display it snapped, If there's not enough space then shrink the playlist to fit and snap, If it still cannot fit then overlay it over MPDN's window on the right.

http://s7.postimg.org/pn7d7thcr/Untitled.jpg

**Comment by Garteal** [comment]
You mean snap by default just like in that screenshot right? Well... I don't know... I personally like how it doesn't cover up the whole screen by default. If you like to have that however you can make use of the "remember window position and size" option.

**Comment by Ryrynz** [comment]
Nah, snapped directly to the left not overlaying. A better idea I think might be to adjust MPDN's window size to accommodate displaying the playlist if MPDN is too large.

**Comment by Ryrynz** [comment]
http://i.imgur.com/asrvfcW.jpg snapped to the right side, by default the playlist just floats near MPDN until the position is saved.

**Comment by Garteal** [comment]
Yeah I dig having it floated, but I'll play with the snapping and see how it'll look.

**Comment by Garteal** [comment]
That can work I guess. I'll push this. Let me know if it needs changing. I made it slightly smaller than a normal docked window since MPDN starts in the center of your screen by default. Now it barely touches it.

**Comment by Ryrynz** [comment]
Sweet, I think we're almost there.

**Comment by Ryrynz** [comment]
The playlist is overlapping over part of the edge of the MPDN window and the playlist displays taking up the whole height of the desktop. Ideally match the height of the MPDN window and touch the windows so there's only 1 pixel of black between them.

**Comment by Garteal** [comment]
Bah misref'd. Should be done in d47443d.

**Comment by Ryrynz** [comment]
Cool, position and scaling works a treat. Now we need it stay in that same snapped position on resize.
I think having that as a single option would be fine, I can't imagine anyone would want the playlist to resize if it's not snapped beside MPDN. Simply call it 'Snap and scale with player'

Also it would be nice to have a lock window size option so that it only scales with the player window

Could probably move those window options into their own category as well.

**Comment by Garteal** [comment]
Check it out and let me know if there needs to be more changes.

**Comment by Ryrynz** [comment]
The window is overlapping and is too high. I've unticked remember window size and also position so this picture shows the default snapped position.

http://i.imgur.com/W19myas.jpg

**Comment by Garteal** [comment]
Yeah... I noticed this. Only happens in the DialogFixed FormBorderStyle, so it's related to that. Gotta do some digging as to why it behaves like that.

**Comment by Garteal** [comment]
Added a workaround for this. And there was a bug where you couldn't resize the window regardless of locking it or not. It should be fixed in 27fc3151af7fe9ff3c1a9c3e288e51ef698da8a7.

**Comment by Ryrynz** [comment]
Does locking the window have to result in the border becoming thicker? The playlist also shifts slightly causing an overlap.

**Comment by Garteal** [comment]
Not seeing thicker borders here on 8.1. Neither do I get an overlap. The current 'workaround' isn't really good anyway since you can have different borders if you customize and it'll have a hardcoded gap.

**Comment by Ryrynz** [comment]
I'm running 8.1 on another PC and I'm seeing the exact same thing on that machine too. Borders thicken and playlist shifts slightly.

**Comment by Garteal** [comment]
Could you give it a try now?

**Comment by Ryrynz** [comment]
Didn't change a thing. :)

**Comment by Garteal** [comment]
Odd... how about now? Are your borders still getting thicker? I've compared mine with screenshots in both cases (then and now) and neither of them showed an increased border size.

**Comment by Ryrynz** [comment]
No more border thickness now. Is it possible to lock the window without the playlist seemingly "jump" or is that just a side effect of the required redraw?

**Comment by Ryrynz** [comment]
Tick snap and scale and lock window and resize MPDN. Now untick snap and scale and lock window. The playlist window loses it's snap to MPDN. Ideally we want the window to remain in the same position  as before.

**Comment by Ryrynz** [comment]
Tick snap and scale and tick remember window position  Launch a file and have MPDN enter fullscreen and the playlist will move to the far top left of the screen.

**Comment by Ryrynz** [comment]
Without remember window position enabled I can move the playlist window and close it and it's position is remembered. Ideally when snap and scale and remember window position are both ticked the playlist only remembers it's position during that MPDN session. Once MPDN is restarted the playlist will snap.

Currently when I tick snap and scale and remember window position and then close and reload MPDN and click view playlist the playlist is snapped But hasn't matched the height of the MPDN window. Making any changes in the playlist options at this point causes the playlist to scale properly and matching MPDN's window.

**Comment by Garteal** [comment]
Thanks a lot for the testing!

https://github.com/Garteal/MPDN_Extensions/issues/2#issuecomment-104854018
The reason the window "jumps" is because the Sizeable FormBorderStyle has a different size than a FixedSingle FormBorderStyle and I have to correct the position of the playlist otherwise it'd be wrongly positioned.

https://github.com/Garteal/MPDN_Extensions/issues/2#issuecomment-104854151
Happens because of the different sizes of Sizable and FixedSingle. And because snapping is disabled it can't correct the position. Disabling one after another does the trick for now but I'll take a further look into it later.

https://github.com/Garteal/MPDN_Extensions/issues/2#issuecomment-104854912
Yeah that's because the remembered window position has a higher precedence than the snapping... should that be reversed? Done that anyway, but let me know.

https://github.com/Garteal/MPDN_Extensions/issues/2#issuecomment-104855198
I forgot to add the snapping to the initializing. That's been done now and it should snap and scale properly on starting up MPDN.

Right now on start-up, the snapping has a higher precedence than the "remember window position" option. I'm not sure why you want them both enabled at the same time, but since it's possible we'll need to have a proper way of handling this.

**Comment by Ryrynz** [comment]
The playlist now moves from it's snapped position when locking, it didn't do that before.
